unidist/core/backends/mpi/core/controller/object_store.py

```diff
@@ -196,10 +196,10 @@ def generate_output_data_id(self, dest_rank, gc, num_returns=1):
         elif num_returns == 0:
             output_ids = None
         else:
-            output_ids = []
-            for _ in range(num_returns):
+            output_ids = [None] * num_returns
```

Comment:
We use many .append calls in communcation, for instance. Let's fix all the places. Also, please take a look at other python code snippets that might be improved.

Reply:
> We use many .append calls in communcation, for instance. Let's fix all the places

Of course, if it gives us benefit.

Reply:
Could find only 2 appends that could be replaced with inserting at list indexes, 

Few other places use conditions to check and then append having a python list of fixed size would not work for these cases , tried list comprehensions with ":=" operator for these as in this commit
https://github.com/modin-project/unidist/pull/290/commits/d965f071aa4eb30893b1b1d354321329f499d500#diff-38853ac231026c17efa13d47a868b3d3ef66be4d2e29e3e2ddc58539423b422eR92
But python<3.8 does not support ":=" so removed those.

Reply:
One of the examples in communication. https://github.com/modin-project/unidist/blob/edbaadf80d5911d3723268950a42b776282448dc/unidist/core/backends/mpi/core/communication.py#L272

Reply:
Please try to replace appends to list indexing check the perf once again on our benchmarks.